[Security Solutions] Sync EA transform with risk engine

machadoum opened the pull request:
## Summary

@elastic/security-entity-analytics

When the risk engine is enabled, users must wait at least one hour to see risk score data.
This PR fixes the issue by scheduling the latest transform after the risk score task finishes.

The first time the risk score runs, it will call `start transform`. That is necessary to ensure the transform doesn't run on an empty index. On the subsequent runs, it will call `schedule transform now`. 

How to test it?
* Open kibana with an empty ES
* Make sure you have the appropriate license
* Ingest some events and generate alerts
* Enable the risk engine
* Open the Entity Analytics Dashboard 
* You should see risk score data (if it isn't there yet, refresh the page)

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


<!--ONMERGE {"backportTargets":["8.11"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

rylnd commented on x-pack/plugins/security_solution/server/lib/risk_engine/configurations.ts:
```diff
@@ -149,9 +149,6 @@ export const getIndexPatternDataStream = (namespace: string): IIndexPatternStrin
   alias: `${riskScoreBaseIndexName}.${riskScoreBaseIndexName}-${namespace}`,
 });
 
-export const getLatestTransformId = (namespace: string): string =>
```
👍 Thank you for taking the time to move this to a less surprising location!

rylnd commented on x-pack/plugins/security_solution/server/lib/risk_engine/tasks/risk_scoring_task.test.ts:
```diff
@@ -491,6 +507,20 @@ describe('Risk Scoring Task', () => {
           }
         });
 
+        it('does not schedules the transform to run now', async () => {
```
It might be useful to add a unit test showing that we do schedule the transform in the case where the task was cancelled 👍

machadoum replied:
Good call!

rylnd commented on x-pack/plugins/security_solution/server/lib/risk_engine/utils/transforms.test.ts:
```diff
@@ -0,0 +1,58 @@
+      state: 'started',
+    },
+  ],
+} as unknown as TransformGetTransformStatsResponse;
```
It doesn't appear that the `as unknown` is necessary here: 
```suggestion
} as TransformGetTransformStatsResponse;
```

ditto below.

rylnd commented on x-pack/plugins/security_solution/server/lib/risk_engine/utils/transforms.test.ts:
```diff
@@ -0,0 +1,58 @@
+      expect(esClient.transform.startTransform).toHaveBeenCalled();
+    });
+
+    it('calls startTransform when the transform state is started ', async () => {
```
```suggestion
    it('calls scheduleNowTransform when the transform state is started ', async () => {
```

rylnd commented on x-pack/plugins/security_solution/server/lib/risk_engine/utils/transforms.ts:
```diff
@@ -110,13 +118,51 @@ export const startTransform = async ({
+    );
+  }
+
+  if (!hasTransformStarted(transformStats.transforms[0])) {
+    await esClient.transform.startTransform({
+      transform_id: transformId,
+    });
+  } else {
+    await esClient.transform.scheduleNowTransform({
+      transform_id: transformId,
+    });
+  }
+};
```
```suggestion
  if (hasTransformStarted(transformStats.transforms[0])) {
    await esClient.transform.scheduleNowTransform({
      transform_id: transformId,
    });
  } else {
    await esClient.transform.startTransform({
      transform_id: transformId,
    });
  }
};
```

rylnd commented on x-pack/test/detection_engine_api_integration/security_and_spaces/group10/risk_engine/risk_scoring_task_execution.ts:
```diff
@@ -109,6 +109,16 @@ export default ({ getService }: FtrProviderContext): void => {
+              transform_id: 'risk_score_latest_transform_default',
+            });
+
+            expect(transformStats.transforms[0].state).to.eql('started');
```
👍

rylnd reviewed (approved):
A few nits, but nothing blocking this from being merged.

Great work here, thanks @machadoum!

machadoum commented:
Thank the sharp review! 🙇‍♂️ 
I addressed all the comments on this commit:  https://github.com/elastic/kibana/pull/167371/commits/7d805abe8f851c054b3481071ec97d2204a562a5.

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.11|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/168345">](https://github.com/elastic/kibana/pull/168345)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
